crates/duty-tracker/src/predicates.rs

```diff
@@ -69,14 +72,26 @@ pub(crate) fn deposit_request_info(
             Some((recovery_x_only_pk, el_addr))
         })?;
 
-    Some(DepositInfo::new(
+    let deposit_info = DepositInfo::new(
         OutPoint::new(tx.compute_txid(), 0),
         stake_index,
         el_addr.to_vec(),
         deposit_request_output.value,
         recovery_x_only_pk,
         deposit_request_output.script_pubkey.clone(),
-    ))
+    );
+
+    // Regenerate the P2TR address from the OP_RETURN data, for now the spend info does all the
+    // necessary validations.
+    deposit_info
+        .compute_spend_infos(build_context, pegout_graph_params.refund_delay)
+        .map_err(|e| {
+            warn!("failed to compute spend info: {:?}", e);
```

Comment:
Should log the txid while we'e at it.

```suggestion
            warn!(err=%e, txid=%tx.compute_txid(), "failed to compute spend info, malformed DRT");
```